main/social/personal_data.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/* For licensing terms, see /license.txt */
+
+/**
+ * @package chamilo.messages
+ */
+$cidReset = true;
+
+require_once __DIR__.'/../inc/global.inc.php';
+use Chamilo\UserBundle\Entity\User;
+use Chamilo\UserBundle\Repository\UserRepository;
+use Chamilo\CoreBundle\Entity\Repository\LegalRepository;
+
+api_block_anonymous_users();
+if (!api_get_configuration_value('enable_gdpr')) {
+    api_not_allowed(true);
+}
+
+$userId = api_get_user_id();
+
+$entityManager = Database::getManager();
+/** @var UserRepository $repository */
+$repository = $entityManager->getRepository('ChamiloUserBundle:User');
```

Comment:
You can use "api_get_user_entity($userId)", There are also "api_get_course_entity()" and "api_get_session_entity() "functions.